We use asm symbol redefining to allow some of legacysupport's magic to work
in certain circumstances this errors, always (so far) involving the <assert.h>
header. work around this by allowing asm symbol redefining again on < 10.6
where this feature is used at present, until/unless we can come up with some more
elegant solution to the problem that doesn't require this.

see https://trac.macports.org/ticket/58677 for further details


--- a/tools/clang/lib/Sema/SemaDecl.cpp.orig	2021-09-19 10:11:38.000000000 -0700
+++ b/tools/clang/lib/Sema/SemaDecl.cpp	2021-09-19 10:16:42.000000000 -0700
@@ -2634,8 +2634,8 @@
     if (AsmLabelAttr *OldA = Old->getAttr<AsmLabelAttr>()) {
       if (OldA->getLabel() != NewA->getLabel()) {
         // This redeclaration changes __asm__ label.
-        Diag(New->getLocation(), diag::err_different_asm_label);
-        Diag(OldA->getLocation(), diag::note_previous_declaration);
+        // Diag(New->getLocation(), diag::err_different_asm_label);
+        // Diag(OldA->getLocation(), diag::note_previous_declaration);
       }
     } else if (Old->isUsed()) {
       // This redeclaration adds an __asm__ label to a declaration that has
